pulsar/internal/connection.go

```diff
@@ -744,6 +745,50 @@ func (c *connection) handleAuthChallenge(authChallenge *pb.CommandAuthChallenge)
 	c.writeCommand(baseCommand(pb.BaseCommand_AUTH_RESPONSE, cmdAuthResponse))
 }
 
+func (c *connection) handleSendError(sendError *pb.CommandSendError, cmdError *pb.CommandError) {
+	c.log.Warnf("Received send error from server: [%v] : [%s]", sendError.GetError(), sendError.GetMessage())
+
+	requestID := cmdError.GetRequestId()
+	producerID := sendError.GetProducerId()
+
+SendError:
+	switch *sendError.Error {
```

Comment:
We also handle the `ChecksumError` in the java implementation, do we need to implement that?

Reply:
This will be supported in the next pr